### ISSUE TITLE
Is there a way to access firebase doc of a specific collection with firebaseConnect dynamically? 

### ISSUE THREAD

**Issue by morpheus86** [prescottprue/redux-firestore#158]
**Do you want to request a *feature* or report a *bug*?**  
request a feature

**What is the current behavior?**
i cannot access my doc in order to get to my nested collection

**What is the expected behavior?**

**Which versions of dependencies, and which browser are affected by this issue? Did this work in previous versions or setups?**

**Steps to reproduce and if possible a minimal demo of the problem via [codesandbox](https://codesandbox.io/) or similar.**
import React from "react";
import { Link } from "react-router-dom";
import { Redirect } from "react-router-dom";
import { connect } from "react-redux";
import { firestoreConnect } from "react-redux-firebase";
import { compose } from "redux";

const FavGems = props => {
  console.log(props);
  const { auth, favorite } = props;
  if (!auth.uid) return <Redirect to="/signin" />;
  return (
    <div className="project-list section">
      {favorite &&
        favorite.map((fav, idx) => {
          return (
            <Link to={"/project/" + idx} key={idx}>
              <div className="card z-depth-0 project-summary">
                <div className="card-content grey-text text-darken-3">
                  <span className="card-title">{fav.name}</span>
                  <p>created by {fav.authors}</p>
                  <p className="grey-text">{fav.info}</p>
                </div>
              </div>
            </Link>
          );
        })}
    </div>
  );
};

const mapState = state => {
  console.log(state);
  return {
    auth: state.firebase.auth,
    favorite: state.firestore.ordered.favorites
  };
};

export default compose(
  connect(mapState),
  firestoreConnect([{ collection: "users" }])
)(FavGems);
// I want to expose a nested collection that is in the user collection but i need to access the the doc which is also the uid. Is there a way for me to grab that ?
Any help would be appreciated

**Comment by prescottprue** [comment]
There are a few things here, so I will separate them:

### Dynamic Listener Config
Both `firebaseConnect` and `firestoreConnect` allow you to pass either an array of listener configs **or** a function that returns an array of listener settings. Like so:

```js
firestoreConnect((props) => [
  `todos/${props.todoId}`
])
```

This is shown in the state based query example

### Query based on UID
This takes connecting to state and passing the state value down as a prop. You will also want to wait for things to load since auth isn't there on initial load. Both of these things are covered [in the state based query example snippet](https://github.com/prescottprue/react-redux-firebase/tree/master/examples/snippets/stateBasedQuery).

**App.js**
```js
const Home = ({ auth }) => {
  // handle initial loading of auth
  if (!isLoaded(auth)) {
    return <div>Loading...</div>
  }

  // auth is empty (i.e. not logged in)
  if (isEmpty(auth)) {
    return <LoginView />
  }

  // auth passed down to TodosView
  return <Home uid={auth.uid} />
}

const enhance = connect(
  ({ firebase: { auth } }) => ({ auth })
)(Home)
```

**Home.js**

```js
const enhance = compose(
  firestoreConnect(({ uid }) => [ // uid comes from props
    {
      collection: 'users',
      doc: uid,
      subcollections: [{ collection: 'favorites' }],
      storeAs: `${uid}-favorites` // place under a unique store key
    }
  ]),
  connect((state) => {
    return {
      favorites: state.firestore.ordered[`${uid}-favorites`]
    }
  })
)
```

Note: You wouldn't need `storeAs` if you were using a `v1.*.*` version from the `alpha` branch, but that seemed out of scope to go over.

**Comment by morpheus86** [comment]
Thank you Scott. I appreciate the quick reoly.    On Monday, December 17, 2018, 11:59:01 PM EST, Scott Prue <notifications@github.com> wrote:  
 
 
There are a few things here, so I will separate them:

Dynamic Listener Config

Both firebaseConnect and firestoreConnect allow you to pass either an array of listener configs or a function that returns an array of listener settings. Like so:
firestoreConnect((props) => [
  `todos/${props.todoId}`
])
This is shown in the state based query example

Query based on UID

This takes connecting to state and passing the state value down as a prop. You will also want to wait for things to load since auth isn't there on initial load. Both of these things are covered in the state based query example snippet.

App.js
const Home = ({ auth }) => {
  // handle initial loading of auth
  if (!isLoaded(auth)) {
    return <div>Loading...</div>
  }

  // auth is empty (i.e. not logged in)
  if (isEmpty(auth)) {
    return <LoginView />
  }

  // auth passed down to TodosView
  return <Home uid={auth.uid} />
}

const enhance = connect(
  ({ firebase: { auth } }) => ({ auth })
)(Home)
Home.js
const enhance = compose(
  firestoreConnect(({ uid }) => [ // uid comes from props
    {
      collection: 'users',
      doc: uid,
      subcollections: [{ collection: 'favorites' }],
      storeAs: `${uid}-favorites` // place under a unique store key
    }
  ]),
  connect((state) => {
    return {
      favorites: state.firestore.ordered[`${uid}-favorites`]
    }
  })
)
Note: You wouldn't need storeAs if you were using a v1.*.* version from the alpha branch, but that seemed out of scope to go over.

**Comment by prescottprue** [comment]
Going to close since this is more of a question and it seems to be answered

**Comment by morpheus86** [comment]
Yes please close it as you were very helpful. 
Thanks again

Sent from Yahoo Mail on Android 
 
  On Tue, Dec 18, 2018 at 20:59, Scott Prue<notifications@github.com> wrote:   
Closed #158.
